zio-schema/shared/src/main/scala/zio/schema/Diff.scala

```diff
@@ -254,6 +258,195 @@ object Differ {
+
+      case (Schema.Tuple(leftSchema, rightSchema), Diff.Tuple(leftDiff, rightDiff)) => {
+        patch(leftSchema, leftDiff) -> patch(rightSchema, rightDiff) match {
+          case (Left(e1), Left(e2)) => Left(s"Errors: $e1 and $e2")
+          case (_, Left(e))         => Left(e)
+          case (Left(e), _)         => Left(e)
+          case (Right(f1), Right(f2)) =>
+            Right((a: A) => {
+              f1(a._1) -> f2(a._2) match {
+                case (Left(e1), Left(e2)) => Left(s"Errors: $e1 and $e2")
+                case (_, Left(e))         => Left(e)
+                case (Left(e), _)         => Left(e)
+                case (Right(l), Right(r)) => Right((l, r))
+              }
+            })
+        }
+      }
+
+      case (Schema.Sequence(schema, fromChunk, toChunk), Diff.Sequence(diffs)) =>
+        Right { (a: A) =>
+          diffs
+            .zipAll(toChunk(a))
+            .flatMap {
+              case (Some(Diff.Total(right, Diff.Tag.Right)), _) => Some(Right(right))
+              case (Some(Diff.Total(_, Diff.Tag.Left)), _)      => None
+              case (Some(diff), Some(value))                    => Some(patch(schema, diff).flatMap(f => f(value)))
+              case (None, Some(value))                          => Some(Left(s"Diff missing for value=$value."))
+              case (Some(diff), None)                           => Some(Left(s"Value missing for diff=$diff."))
+              case (None, None)                                 => Some(Left(s"Unknown error in sequence."))
+            }
+            .partitionMap(identity) match {
+            case (Chunk.empty, values) => Right(fromChunk(values))
+            case (errors, _)           => Left(s"Running patch produced the following error(s): ${errors.toList}.")
+          }
+        }
+
+      // TODO - how do you tell if diff is from left or right?
+      case (Schema.EitherSchema(leftSchema, rightSchema @ _), diff) =>
+        patch(leftSchema, diff).map(f => (a: A) => Right(a.map(f(_))))
+
+      case (Schema.Transform(schema, f, g), diff) =>
+        Right { (a: A) =>
+          for {
+            to      <- g(a)
+            patchF  <- patch(schema, diff)
+            backToA <- patchF(to).flatMap(f)
+          } yield backToA
+        }
+
+      case (Schema.GenericRecord(structure: FieldSet), Diff.Record(diffs: ListMap[String, Diff])) =>
+        Right { (a: A) =>
+          val values: ListMap[String, DynamicValue] = schema.toDynamic(a) match {
+            case DynamicValue.Record(values) => values
+          }
+          patchProductData(structure.toChunk, values, diffs).asInstanceOf[Either[String, A]]
+        }
+
+      case (schema: Schema.Record[A], Diff.Record(diffs: ListMap[String, Diff])) =>
+        Right { (a: A) =>
+          val values: ListMap[String, DynamicValue] = schema.toDynamic(a) match {
+            case DynamicValue.Record(values) => values
+          }
```

Comment:
Should probably do a wildcard case here just in case. It should in theory never happen but don't want to fail with a MatchError in case it does.